logging/slog.go

```diff
@@ -0,0 +1,64 @@
+package logging
+
+import (
+	"context"
+
+	"log/slog"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+// TraceIDKey is the key used by loggers for the trace ID field in key/value pairs.
+// It is set as a variable rather than a constant so that it can be changed by users at startup.
+var TraceIDKey = "traceID"
+
+// NewSLogLogger creates a new SLogLogger which wraps an *slog.Logger that has a handler to always add a trace ID
+// to the log messages if the context is provided in the log call (e.g. InfoContext())
+func NewSLogLogger(handler slog.Handler) *SLogLogger {
+	return &SLogLogger{
+		Logger: *slog.New(&traceIDHandler{next: handler}),
+	}
+}
+
+// SLogLogger wraps slog.Logger to override the With() method to return a Logger (*SLogLogger), rather than *slog.Logger,
+// implementing the Logger interface.
+type SLogLogger struct {
+	slog.Logger
```

Comment:
Not sure if it's a good idea to wrap a value here, how about we wrap a pointer – this will also make New and With less awkward.
```suggestion
	*slog.Logger
```